laboratory/src/pages/with-ethereum-provider/index.tsx

```diff
@@ -36,11 +38,19 @@ export default function WithEthereumProvider() {
   }
 
   async function onDisconnect() {
-    if (providerClient) {
-      await providerClient.disconnect()
-      setSession(false)
-    } else {
-      throw new Error('providerClient is not initialized')
+    if (!disconnecting) {
+      if (providerClient) {
+        setDisconnecting(true)
+        try {
+          await providerClient.disconnect()
+        } catch (error) {
+          showToast.error('Something went wrong', { duration: 2000 })
```

Comment:
1. Let's `console.error` here as well with full error object
2. If possible toast should try to show `error.message` and then fallback to generic "Something went wrong"